client/mysqladmin.cc

```diff
@@ -373,6 +373,11 @@ int main(int argc,char *argv[])
   int error= 0, temp_argc;
   MYSQL mysql;
   char **commands, **save_argv, **temp_argv;
+  char *tmp;
+  
+  tmp= (char *) getenv("MYSQL_HOST");
+  if (tmp && !host)
+    host= my_strdup(PSI_NOT_INSTRUMENTED, tmp, MYF(MY_WME));
```

Comment:
why are you doing a strdup? Is an assignment enough? This memory isn't `free`d so if a `host= tmp` works can we use that.

Reply:
This was copy and pasted from mysql.cc https://github.com/MariaDB/server/blob/11.0/client/mysql.cc#L2135-L2137. I will try your suggestion and see if it works or free up the memory